### PR TITLE
ci ubuntu 20

### PR DESCRIPTION
**Pull request by turuslan**
- update hunter, libp2p, ubuntu
- remove unused
- fix tests in release

### CODE REVIEW

File: core/sector_storage/stores/impl/index_impl.cpp [hunk 1]
```diff
@@ -342,9 +342,8 @@ namespace fc::sector_storage::stores {
     std::sort(candidates.begin(),
               candidates.end(),
               [](const StorageEntry &lhs, const StorageEntry &rhs) {
```
Comment (turuslan) [inline core/sector_storage/stores/impl/index_impl.cpp:344]: sector_index_test failed in release build on these lines

File: test/core/markets/retrieval/fixture.hpp [hunk 2]
```diff
@@ -151,9 +155,7 @@ namespace fc::markets::retrieval::test {
           host, datatransfer, api, client_ipfs);
       provider->start();
```
Comment (turuslan) [inline test/core/markets/retrieval/fixture.hpp:157]: `addPieceSample` never called in release build

File: core/common/soralog.hpp [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <libp2p/log/configurator.hpp>
+#include <libp2p/log/logger.hpp>
+
+namespace fc {
+  inline void libp2pSoralog() {
```
Comment (Alexey-N-Chernyshov) [inline core/common/soralog.hpp:12]: Please, add comment that it is a log for libp2p only. I think `core/common/libp2p/` is a better place for it.

**Review by Alexey-N-Chernyshov — APPROVED** [review]

**Review by Alexey-N-Chernyshov — APPROVED** [review]

File: core/blockchain/block_validator/impl/block_validator_impl.cpp [hunk 4]
```diff
@@ -150,9 +149,8 @@ namespace fc::blockchain::block_validator {
 
   outcome::result<std::reference_wrapper<BlockValidatorImpl::TipsetCPtr>>
   BlockValidatorImpl::getParentTipset(const BlockHeader &block) const {
-    IPLDBlock ipld_block = IPLDBlock::create(block);
-    if (parent_tipset_cache_
-        && parent_tipset_cache_.value().first == ipld_block.cid) {
+    auto cid{primitives::cid::getCidOfCbor(block).value()};
```
Comment (ortyomka) [inline core/blockchain/block_validator/impl/block_validator_impl.cpp:152]: Maybe check to error?

**Review by ortyomka — APPROVED** [review]

File: core/blockchain/block_validator/impl/block_validator_impl.cpp (hunk 4 above)
Comment (Alexey-N-Chernyshov) [reply core/blockchain/block_validator/impl/block_validator_impl.cpp:152]: Or add assert that there is no error.

Comment (turuslan) [reply core/blockchain/block_validator/impl/block_validator_impl.cpp:152]: `.value()` throws `std::system_error` (like OUTCOME_EXCEPT or cbor streams)

**Pull request merged**